src/AWS.Deploy.CLI/AWSUtilities.cs

```diff
@@ -22,59 +23,75 @@ public AWSUtilities(IToolInteractiveService toolInteractiveService)
 
         public async Task<AWSCredentials> ResolveAWSCredentials(string profileName, string lastUsedProfileName)
         {
-            AWSCredentials credentials;
 
-            var chain = new CredentialProfileStoreChain();
 
-            if (!string.IsNullOrEmpty(profileName))
+            async Task<AWSCredentials> Resolve()
             {
-                if (chain.TryGetAWSCredentials(profileName, out credentials) &&
+                AWSCredentials credentials;
+
+                var chain = new CredentialProfileStoreChain();
+
+                if (!string.IsNullOrEmpty(profileName))
+                {
+                    if (chain.TryGetAWSCredentials(profileName, out credentials) &&
+                        // Skip checking CanLoadCredentials for AssumeRoleAWSCredentials because it might require an MFA token and the callback hasn't been setup yet.
+                        (credentials is AssumeRoleAWSCredentials || await CanLoadCredentials(credentials)))
```

Comment:
I think you can combine the nested ifs to one like you did below L45.

Reply:
Done